DQM/HGCal/plugins/HGCalTriggerClient.cc

```diff
@@ -84,103 +204,434 @@ void HGCalTriggerClient::analyze(const edm::Event& iEvent, const edm::EventSetup
+    // 	}else{
+    // 	  nofBxCentralMM1++;
+    // 	  isBxCE1 = true;
+    // 	}
+    //   }
+    // }else{
+    //   if((ev.bxId%8 != bx_raw[iect][daq_nbx[0]][0]) || (ev.bxId%8 != bx_unpkd[iect][daq_nbx[1]][0])){
+    // 	//std::cerr << "Bx Module test failed for iect : "<< iect <<" since ev.bxId%8 : " << (ev.bxId%8) <<" and bx_raw : "<< bx_raw[iect][daq_nbx[0]][0] << " and bx_unpkd : "<< bx_unpkd[iect][daq_nbx[1]][0] << std::endl ;
+    // 	isGood = false;
+    // 	if(iect==0) {
+    // 	  nofBxCentralMM++;
+    // 	  isBxCE = true;
+    // 	}else{
+    // 	  nofBxCentralMM1++;
+    // 	  isBxCE1 = true;
+    // 	}
+    //   }
+    // }
+  }
+
+  //Check Locations
+  for (int iect = 0; iect < 2; iect++) {
+    for (int ibx = 0; ibx < maxnbx; ibx++) {
+      for (int istc = 0; istc < 12; istc++) {
+        if ((loc_unpkd[iect][ibx][istc] >> 2 & 0xF) != unsigned(istc)) {
+          if (verbose_ >= 4)
+            std::cerr << " Unpacked location index do not match with the STC for (Run, event, iecont, bx, stc, "
+                         "istc_from_unpacked) : "
+                      << "(" << runNumber_ << "," << iEvent.id().event() << "," << iect << "," << ibx << "," << istc
+                      << "," << (loc_unpkd[iect][ibx][istc] >> 2 & 0xF) << ") " << std::endl;
+          if (iect == 0) {
+            isSTCNE0 = true;
+          } else {
+            isSTCNE1 = true;
+          }
+        }
+        if ((loc_unpkd[iect][ibx][istc] & 0x3) != loc_raw[iect][ibx][istc]) {
+          if (verbose_ >= 4)
+            std::cerr << " Unpacked location value do not match with the packed one for (Run, event, iecont, bx, stc, "
+                         "loc_unpacked, loc_packed ) : "
+                      << "(" << runNumber_ << "," << iEvent.id().event() << "," << iect << "," << ibx << "," << istc
+                      << "," << (loc_unpkd[iect][ibx][istc] & 0x3) << "," << loc_raw[iect][ibx][istc] << ") "
+                      << std::endl;
+          if (iect == 0) {
+            isSTCLE0 = true;
+          } else {
+            isSTCLE1 = true;
+          }
+        }
+      }
+    }
+  }
+
+  //Check Energy
+  for (int iect = 0; iect < 2; iect++) {
+    for (int ibx = 0; ibx < maxnbx; ibx++) {
+      for (int istc = 0; istc < 12; istc++) {
+        if (energy_raw[iect][ibx][istc] != energy_unpkd[iect][ibx][istc]) {
+          if (verbose_ >= 4)
+            std::cerr << " Packed and unpacked energies does not match for (Run, "
+                         "event,iecont,bx.stc,raw_energy,unpacked_energy) : "
+                      << "(" << runNumber_ << "," << iEvent.id().event() << "," << iect << "," << ibx << "," << istc
+                      << "," << energy_raw[iect][ibx][istc] << "," << energy_unpkd[iect][ibx][istc] << ") "
+                      << std::endl;
+
+          if (iect == 0) {
+            isEngE0 = true;
+          } else {
+            isEngE1 = true;
+          }
+        }
+      }
+    }
+  }
+
+  if (isSTCNE0)
```

Comment:
Same comment as above on `switch`.

Reply:
These are different than the switch/case for L1a trigger type. The trigger type is unique for a given event. However, more than one errors of STC location, indices, bx or energy can be present for a given event. So if switch is used for one type others will not be checked. Therefore it is not applied at present. I press the "resolve" button below. Please reopen it if I have missed the point.

Reply:
I see your point; I was mislead by the code structure.

Imagine next time you want to add a new one bin to that histogram: you'd have to add something in 3 places.
In this sort of case, I'd have a `map<string, int>`: the keys are all the errors you declare and then you can increment and fill in an automated way.